src/ai/NetworkInterface.js

```diff
@@ -0,0 +1,136 @@
+import { credentials } from './APIKeys.js';
+
+/**
+ * @class NetworkInterface
+ * Provides a central location to manage the various fetches made to the toolbox edge server to interact with the
+ * /ai routes on the server, which in turn interact with Azure/OpenAI APIs.
+ * 
+ * @todo: the fetch paths have been temporarily adjusted to work with localhost and locally provided keys,
+ *   and the requests may need to be changed again if you are working with the cloud proxy
+ */
+export class NetworkInterface {
+    constructor() {
+        this.postAiApiKeys(credentials.gpt4o.endPoint, credentials.gpt4o.apiKey, true).then(res => {
+            console.log(`AI API Keys posted: ${res}`);
+        }).catch(err => {
+            console.warn(`Error posting AI API keys: ${err}`);
+        });
+    }
+
+    /**
+     * Note: this has been added for developing on localhost.
+     * On cloud proxy, API keys are provided on backend in another way.
+     */
+    postAiApiKeys(endpoint, azureApiKey, isInit = false) {
+        return new Promise(async (resolve, reject) => {
+            if (!endpoint || !azureApiKey) {
+                reject(`Can't post API keys because endpoint or API key is invalid: ${endpoint}, ${azureApiKey}`);
+            }
+
+            let worldId = await this.findWorldId();
+            let ip = worldId.ip;
+            let port = realityEditor.network.getPort(worldId);
+            let route = '/ai/init';
+
+            realityEditor.network.postData(realityEditor.network.getURL(ip, port, route),
+                {
+                    endpoint: endpoint,
+                    azureApiKey: azureApiKey,
+                },
+                function (err, res) {
+                    if (err) {
+                        console.warn();
+                        reject(`/ai/init error: ${err}`);
+                    } else {
+                        if (res.answer === 'success') {
+                            // change ai search text area to the actual search text area
+                            console.log('ai init success');
+                            realityEditor.ai.hideEndpointApiKeyAndShowSearchTextArea();
+                            if (isInit) {
+                                // todo Steve: broadcast this message to all avatars, and have them spin up their own Azure GPT-3.5 with the same API keys
```

Comment:
If we share API keys within the session and users are inputting API keys, we absolutely need to warn users that their API keys will be shared with others in the call.

Reply:
@Steve-KX-RL do we need this API key sharing method anymore?

Reply:
If we're testing the local AI apis, we would still need them